### PR TITLE
Create YAML specification

### PR DESCRIPTION
**Pull request by mak2508**

### CODE REVIEW

**Comment by uabbas** [comment]
You need to install the pylint hook. Also add instructions to do this in the README.md as well.

**Comment by uabbas** [comment]
YAML parser needs to support observers as well.

**Comment by mak2508** [comment]
There are a lot of file changes, since I added code formatters to the pre-commit hooks that have formatted many files.

The relevant files that I worked on are the following:
- .pre-commit-config.yaml
- tasrif/yaml_parser.py
- CONTRIBUTING.md
- All the YAML files that specify the different example pipelines
- requirement.txt and qa-requirement.txt (small changes)

There are also the following files, which were mostly written by Ummar. Let me know if you want them deleted (so that it is not merged to master):
- Medium Post.ipynb
- example.yaml
- tasrif/processing_pipeline/map_iterable_operator.py
- tasrif/test_scripts/test_map_iterable_operator.py

There are some other smaller changes as well, such as running the YAML pipelines in the examples. If you want a more exhaustive list, do let me know.

File: entrypoint.py [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
-    'examples/fitbit_intraday/*.py',
-    'examples/my_heart_counts/[!test_]*.py'
-    'examples/siha/*.py',
-    'examples/sleep_health/*.py',
-    'examples/withings/*.py',
-    'examples/zenodo_fitbit/*.py',
-    'tasrif/test_scripts/*.py'
+    "examples/fitbit_interday/*.py",
+    "examples/fitbit_intraday/*.py",
+    "examples/my_heart_counts/[!test_]*.py" "examples/siha/*.py",
```
Comment (uabbas) [inline entrypoint.py:10]: a comma is missing here

File: example.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+modules:
```
Comment (uabbas) [inline example.yaml:1]: Are we going to make this necessary. i thought we discussed that we would automatically include tasrif packages?

File: Medium Post.ipynb [hunk 3]
```diff
@@ -0,0 +1,1014 @@
+{
```
Comment (uabbas) [inline Medium Post.ipynb:1]: Remove this file for the time being until the post is published.

File: example.yaml (hunk 2 above)
Comment (mak2508) [reply example.yaml:1]: I have left this for a separate PR. The plan is to just mention the package, and all the operators would be imported

File: Medium Post.ipynb (hunk 3 above)
Comment (mak2508) [reply Medium Post.ipynb:1]: Should I remove `example.yaml` along with the file?

**Review by uabbas — DISMISSED** [review]

**Pull request merged**